controllers/utils.js

```diff
@@ -33,71 +33,161 @@ function hasUserSignedUpForRelay(user) {
   // we do not enroll them in the experiment.
   if (language) {
+
+    if (!Array.isArray(language)) {
+      throw new Error("The language param is not an array");
+    }
     const lang = req.headers["accept-language"].split(",");
-    if (language && !lang[0].includes(language)) {
-      log.debug(`Preferred language is not ${language} variant: ${lang[0]}`);
+
+    // Check to make sure one of the experiment langauge(s) is the top-preferred language.
+    const firstLangMatch = (element) => lang[0].includes(element);
+    if (language && !language.some(firstLangMatch)) {
+      log.debug(`Preferred language is not [${language}] variant: ${lang[0]}`);
+      session.excludeFromExperiment = true;
+      session.experimentBranch = false;
+      session.treatmentBranch = false;
       return false;
     }
   }
 
   // If URL param has experimentBranch entry, use that branch;
   if (req.query.experimentBranch) {
-    if (!["va", "vb"].includes(req.query.experimentBranch)) {
+    if (!Object.keys(variations).includes(req.query.experimentBranch)) {
       log.debug("The requested branch is unknown: ", req.query.experimentBranch);
+      session.excludeFromExperiment = true;
+      session.experimentBranch = false;
+      session.treatmentBranch = false;
       return false;
     }
     log.debug("This session has been set to the requested branch: ", req.query.experimentBranch);
-    req.session.excludeFromExperiment = false;
-    req.session.experimentBranch = req.query.experimentBranch;
+    session.excludeFromExperiment = false;
+    session.experimentBranch = req.query.experimentBranch;
     return req.query.experimentBranch;
   }
 
   // If user was already assigned a branch, stay in that branch;
-  if (req.session.experimentBranch) {
-    log.debug("This session has already been assigned: ", req.session.experimentBranch);
-    return req.session.experimentBranch;
+  if (session.experimentBranch) {
+    log.debug("This session has already been assigned: ", session.experimentBranch);
+    return session.experimentBranch;
   }
 
-  // Growth Team Experiment 2 only wants to expose 29/29/42 of all site traffic to
-  // the experiment. Of the 58% percent inside the experiment, will be split
-  // 50/50 between treatment and control.
-  if (sorterNum < 29) {
-    log.debug("This session has been randomly assigned to the control group. (va)");
-    req.session.experimentBranch = "va";
-    return "va";
-  } else if (sorterNum > 28 && sorterNum < 58) {
-    log.debug("This session has been randomly assigned to the treatment group. (vb)");
-    req.session.experimentBranch = "vb";
-    return "vb";
+  if (sorterNum === false) {
+    sorterNum = Math.floor(Math.random() * 10000) + 1;
+    sorterNum = sorterNum/100;
+    // sorterNum = Math.floor(Math.random() * 100);
+    log.debug("No coinflip number provided. Coinflip number is ", sorterNum);
+  } else {
+    log.debug("Coinflip number provided. Coinflip number is ", sorterNum);
```

Comment:
This else was added to add the coin flip result to the debug console.

Reply:
Wondering if we really need the `excludeFromExperiment` session var or if maybe it would just be enough to check for `request.session.experimentBranch` and assume that anything without `req.session.experimentBranch` is excluded from the experiment. Or set `req.session.experiment` equal to `null`.